pkg/querier/querier.go

```diff
@@ -103,14 +102,15 @@ func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
 	flagext.DeprecatedFlag(f, "querier.iterators", "Deprecated: Use iterators to execute query. This flag is no longer functional; Batch iterator is always enabled instead.", util_log.Logger)
 	//lint:ignore faillint Need to pass the global logger like this for warning on deprecated methods
 	flagext.DeprecatedFlag(f, "querier.batch-iterators", "Deprecated: Use batch iterators to execute query. This flag is no longer functional; Batch iterator is always enabled now.", util_log.Logger)
+	//lint:ignore faillint Need to pass the global logger like this for warning on deprecated methods
+	flagext.DeprecatedFlag(f, "querier.query-store-for-labels-enabled", "Deprecated: Querying long-term store is always enabled.", util_log.Logger)
```

Comment:
Is this intended?

Reply:
Yeah .. thi give a warning if the flag is still present on the cli.., make sense ?

Reply:
Shouldn't we set this flag when we start to deprecate the flag? I think the title is a bit of misleading...
`Remove query_store_for_labels_enabled configuration` we are actually deprecating the config here. Not removing.

Removing means to remove the flag entirely and it should throw error if specified